lib/rubocop/cop/naming/constant_wording.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Naming
+      # This cop checks whether constant names are written using
+      # clearer wording.
+      #
+      # You can customize the mapping from undesired name to why
+      # this method is not desired.
+      #
+      # e.g. not to use `whitelist`:
+      #
+      #   Naming/ConstantWording:
+      #     PreferredMethods:
+      #       whitelist: 'Please use clearer concepts, such as allow, permitted, approved.'
+      #
+      # @example
+      #   # bad
+      #   Whitelist = []
+      #   WhiteList = []
+      #   Whitelisted = []
+      #   Blacklist = []
+      #   BLACKLISTED = []
+      #
+      class ConstantWording < Cop
+        include MethodPreference
+
+        MSG = 'Prefer `%<prefer>` over `%<current>`.'.freeze
+
+        def on_casgn(node)
+          constant_name = get_constant_name(node)
+
+          check_constant_name(constant_name)
+        end
+
+        private
+
+        def get_constant_name(node)
+          if node.parent && node.parent.or_asgn_type?
+            lhs, _value = *node.parent
+            _scope, const_name = *lhs
+          else
+            _scope, const_name, _value = *node
+          end
+
+          const_name
+        end
+
+        def message(node)
+          constant_name = get_constant_name(node)
+
+          format(MSG,
+                 prefer: preferred_method(constant_name),
+                 current: constant_name)
+        end
+
+        def check_constant_name(constant_name)
+          return unless preferred_methods[constant_name.downcase]
```

Comment:
I am debugging a `NoMethodError` here:

```ruby
    13: def preferred_methods
    14:   @preferred_methods ||=
    15:     begin
    16:       # Make sure default configuration 'foo' => 'bar' is removed from
    17:       # the total configuration if there is a 'bar' => 'foo' override.
 => 18:       default = default_cop_config['PreferredMethods']
    19:       merged = cop_config['PreferredMethods']
    20:       overrides = merged.values - default.values
    21:       merged.reject { |key, _| overrides.include?(key) }.symbolize_keys
    22:     end
    23: end

[1] pry(#<RuboCop::Cop::Naming::ConstantWording>)> default_cop_config
=> {"Description"=>"Constant name should use clearer wording.",
 "Enabled"=>true,
 "VersionAdded"=>"0.60",
 "PreferredMethods"=>{"whitelist"=>"Please use clearer concepts, such as allow, permitted, approved."}}
[2] pry(#<RuboCop::Cop::Naming::ConstantWording>)> cop_config
=> {"Enabled"=>true}
```

The `cop_config['PreferredMethods']` only return `{"Enabled"=>true}` instead of what we specified here

https://github.com/rubocop-hq/rubocop/blob/a29e653b5489c89542672b56a14eed127358a2c3/config/default.yml#L1755-L1759

resulting `merged` became `nil` and `merged.values` raises a `NoMethodError`.

I am probably missing something trivial here.